src/plugins/profiling/server/routes/search_flamechart.ts

```diff
@@ -271,71 +271,75 @@ async function queryFlameGraph(
     logger.info('unique downsampled stacktraces: ' + stackTraceEvents.size);
   }
 
-  const resStackTraces = await logExecutionLatency(
+  const nQueries = 4;
+  const results = new Array(nQueries);
+
+  await logExecutionLatency(
     logger,
     'mget query for ' + stackTraceEvents.size + ' stacktraces',
     async () => {
-      if (testing) {
-        return await client.search(
-          {
+      const functions = new Array(nQueries);
+      const promises = new Array(nQueries);
+      const chunkSize = Math.floor(stackTraceEvents.size / nQueries);
+      const stackTraceIDs = [...stackTraceEvents.keys()];
+
+      logger.info('A');
+
+      for (let i = 0; i < nQueries; i++) {
+        functions[i] = async () => {
+          const chunk = stackTraceIDs.slice(chunkSize * i, chunkSize * (i + 1));
+          return client.mget({
             index: 'profiling-stacktraces',
-            size: stackTraceEvents.size,
-            sort: '_doc',
-            query: {
-              ids: {
-                values: [...stackTraceEvents.keys()],
-              },
-            },
-            _source: false,
-            docvalue_fields: ['FrameID', 'Type'],
-          },
-          {
-            querystring: {
-              filter_path: 'hits.hits._id,hits.hits.fields.FrameID,hits.hits.fields.Type',
-            },
-          }
-        );
-      } else {
-        return await client.mget({
-          index: 'profiling-stacktraces',
-          ids: [...stackTraceEvents.keys()],
-          _source_includes: ['FrameID', 'Type'],
-        });
+            ids: [...chunk],
+            _source_includes: ['FrameID', 'Type'],
+          });
+        };
+
+        // Build and send the queries asynchronously.
+        promises[i] = functions[i]();
+      }
+
+      logger.info('B');
+
+      // Wait for the queries to finish.
+      // How can we wait until the fastest query is back ?
+      for (let i = 0; i < nQueries; i++) {
+        results[i] = await promises[i];
       }
```

Comment:
If you want to stop after one promise finishes first, then you can use [Promise.any](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/any).

For example,
```
await Promise.any(nQueries)
             .then(valueOfFastestQuery => ...)
             .catch(...)
```

Reply:
Thanks, got it working.

Reply:
Yep, this seems to be the simplest thing: creating an array with the slices of chinks, then calling N async functions that returns the promises, and `.map` on them

Reply:
I was too fast... Promise.any doesn't give me control over every single background request. I need the exact timing plus I would like to do some processing on each individual result (while other requests are still running in the background).